libraries/AP_Landing/AP_Landing_Slope.cpp

```diff
@@ -384,7 +384,7 @@ void AP_Landing::type_slope_log(void) const
                                             type_slope_stage,
                                             flags,
                                             type_slope_flags,
-                                            slope,
-                                            initial_slope,
-                                            alt_offset);
+                                            (double)slope,
+                                            (double)initial_slope,
+                                            (double)alt_offset);
```

Comment:
C/C++ rule: float var args are promoted to double. As such the compiler gives warning of  implicit float to double conversion.  This is the reason we cast to double in our printf-like functions.